### PR TITLE
mavproxy_map: add context-menu option for printing google maps link

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: MAVProxy/modules/mavproxy_map/__init__.py [hunk 1]
```diff
@@ -143,6 +144,12 @@ def show_position(self):
         posbox = MPMenuChildMessageDialog('Position', msg, font_size=32)
         posbox.show()
 
+    def print_google_maps_link(self):
+        '''show map position click information'''
+        pos = self.mpstate.click_location
+        dms = (mp_util.degrees_to_dms(pos[0]), mp_util.degrees_to_dms(pos[1]))
+#        print("https://www.google.com/maps/place/%s,%s/@%f,%f,21z" % (dms[0], dms[1], pos[0], pos[1]))
+        print("https://www.google.com/maps/search/?api=1&query=%f,%f" % (pos[0], pos[1]))
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/__init__.py:152]: a bit of whitespace needed below

File: MAVProxy/modules/mavproxy_map/__init__.py [hunk 2]
```diff
@@ -143,6 +144,12 @@ def show_position(self):
         posbox = MPMenuChildMessageDialog('Position', msg, font_size=32)
         posbox.show()
 
+    def print_google_maps_link(self):
+        '''show map position click information'''
+        pos = self.mpstate.click_location
+        dms = (mp_util.degrees_to_dms(pos[0]), mp_util.degrees_to_dms(pos[1]))
+#        print("https://www.google.com/maps/place/%s,%s/@%f,%f,21z" % (dms[0], dms[1], pos[0], pos[1]))
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/__init__.py:151]: remove incorrect line

File: MAVProxy/modules/mavproxy_map/__init__.py [hunk 3]
```diff
@@ -143,6 +144,12 @@ def show_position(self):
         posbox = MPMenuChildMessageDialog('Position', msg, font_size=32)
         posbox.show()
 
+    def print_google_maps_link(self):
+        '''show map position click information'''
+        pos = self.mpstate.click_location
+        dms = (mp_util.degrees_to_dms(pos[0]), mp_util.degrees_to_dms(pos[1]))
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/__init__.py:150]: dms not needed

File: MAVProxy/modules/mavproxy_map/__init__.py [hunk 4]
```diff
@@ -89,6 +89,7 @@ def __init__(self, mpstate):
         self.add_menu(MPMenuItem('Set Origin (with height)', 'Set Origin', '# map setorigin '))
         self.add_menu(MPMenuItem('Terrain Check', 'Terrain Check', '# terrain check'))
         self.add_menu(MPMenuItem('Show Position', 'Show Position', 'showPosition'))
+        self.add_menu(MPMenuItem('Print Google Maps Link', 'Print Google Maps Link', 'printGoogleMapsPosition'))
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/__init__.py:92]: 'Google Maps Link' is better (for narrower screens)

**Review by tridge — CHANGES_REQUESTED** [review]

File: MAVProxy/modules/mavproxy_map/__init__.py (hunk 1 above)
Comment (peterbarker) [reply MAVProxy/modules/mavproxy_map/__init__.py:152]: Done

File: MAVProxy/modules/mavproxy_map/__init__.py (hunk 2 above)
Comment (peterbarker) [reply MAVProxy/modules/mavproxy_map/__init__.py:151]: Done

File: MAVProxy/modules/mavproxy_map/__init__.py (hunk 3 above)
Comment (peterbarker) [reply MAVProxy/modules/mavproxy_map/__init__.py:150]: Fixed

File: MAVProxy/modules/mavproxy_map/__init__.py (hunk 4 above)
Comment (peterbarker) [reply MAVProxy/modules/mavproxy_map/__init__.py]: Fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**